### ISSUE TITLE
Username shown wrong in username field

### ISSUE THREAD

**Issue by tatvarajendra** [AzureAD/azure-activedirectory-library-for-objc#1509]
We are using ADAL version 4.0.4 and below method

[authContext acquireTokenWithResource:strResouceURL clientId:strClientID redirectUri:[NSURL URLWithString:strRedirectURI] promptBehavior:AD_FORCE_PROMPT userIdentifier:nil extraQueryParameters:[[NSString stringWithFormat:@"login_hint=%@",username] stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]] completionBlock:^(ADAuthenticationResult *result) { if (AD_SUCCEEDED != result.status){ // display error on the screen } else{//NSLog(@"ADFS Token : %@",result.accessToken); completionBlock(result.accessToken);}}]

My username is like test\adfsuser
But in ADFS page its showing like test%5Cadfsuser

It was working previously with ADAL version 2.7.7

So it seems some decode issue at the ADAL library. Please let me know as soon as possible.

**Comment by tatvarajendra** [comment]
We want to update the library version, as v2.7.7 is using the UIWebview.
And apple will stop accepting webview in future. 
Else ADAL v2.7.7 seems working fine in xcode 11 and iOS 13.

**Comment by oldalton** [comment]
@tatvarajendra, thanks for reporting this issue. I've applied a fix and it will be available in ADAL 4.0.5 release: https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/1512

In the meantime, you can pass login hint to ADAL directly (which is actually the recommended way as well). Here's the code sample to achieve it:

```
ADUserIdentifier *userIdentifier = [ADUserIdentifier identifierWithId:@"yourloginhint@contoso.com" type:OptionalDisplayableId];
    
    [context acquireTokenWithResource:resource
                             clientId:clientId
                          redirectUri:redirectUri
                       promptBehavior:[self promptBehavior]
                       userIdentifier:userIdentifier
                 extraQueryParameters:extraQueryParameters
                               claims:claims
                      completionBlock:^(ADAuthenticationResult *result)
    {
....
}];
```

**Comment by tatvarajendra** [comment]
Thanks for the quick reply. Its working fine with version 4.0.5. 
Thanks
